addons/mail/static/src/messaging/component/notification_list/notification_list_notification_group_tests.js

```diff
@@ -0,0 +1,569 @@
+    );
+    assert.strictEqual(
+        document.querySelector('.o_NotificationGroup_counter').textContent.trim(),
+        "(2)",
+        "should have 2 notifications in the group"
+    );
+
+    document.querySelector('.o_NotificationGroup').click();
+    assert.verifySteps(
+        ['do_action'],
+        "should have called do_action correctly"
+    );
+});
+
+QUnit.test('different mail.channel are not grouped', async function (assert) {
+    // `mail.channel` is a special case where notifications are not grouped when
+    // they are linked to different channels, even though the model is the same.
+    assert.expect(6);
+
+    this.data.initMessaging.mail_failures = [{
+        date: moment.utc().format("YYYY-MM-DD HH:mm:ss"),
+        id: 11,
+        message_type: 'email',
+        model: 'mail.channel', // key element of this test: `mail.channel`
+        model_name: "Channel",
+        notifications: [{
+            failure_type: 'SMTP',
+            id: 21,
+            notification_status: 'exception',
+            notification_type: 'email',
+            partner_id: [41, "Someone"],
+        }],
+        res_id: 31, // key element of this test: a different res_id
+    }, {
+        date: moment.utc().format("YYYY-MM-DD HH:mm:ss"),
+        id: 12,
+        message_type: 'email',
+        model: 'mail.channel', // key element of this test: `mail.channel`
+        model_name: "Channel",
+        notifications: [{
+            failure_type: 'SMTP',
+            id: 22,
+            notification_status: 'bounce',
+            notification_type: 'email',
+            partner_id: [42, "Someone else"],
+        }],
+        res_id: 32, // key element of this test: a different res_id
+    }];
+    await this.start({
+        hasChatWindow: true, // needed to assert thread.open
+    });
+    await this.createNotificationListComponent();
+
+    assert.containsN(
+        document.body,
+        '.o_NotificationGroup',
+        2,
+        "should have 2 notifications group"
+    );
+    const groups = document.querySelectorAll('.o_NotificationGroup');
+    assert.containsOnce(
+        groups[0],
+        '.o_NotificationGroup_counter',
+        "should have 1 group counter in first group"
+    );
+    assert.strictEqual(
+        groups[0].querySelector('.o_NotificationGroup_counter').textContent.trim(),
+        "(1)",
+        "should have 1 notification in first group"
+    );
+    assert.containsOnce(
+        groups[1],
+        '.o_NotificationGroup_counter',
+        "should have 1 group counter in second group"
+    );
+    assert.strictEqual(
+        groups[1].querySelector('.o_NotificationGroup_counter').textContent.trim(),
+        "(1)",
+        "should have 1 notification in second group"
+    );
+
+    await afterNextRender(() => {
+        document.querySelector('.o_NotificationGroup').click();
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_ChatWindow',
+        "should have opened thread"
```

Comment:
_"... channel of notification failure related to 1st group in a chat window"_

Reply:
ok for the clarification except I don't want to explicit "chat window" because this test really does not care if it's open in chat window or discuss

Reply:
It shouldn't, but at the moment it does rely on that and it's made clear by assertion on `.o_ChatWindow`. I prefer a clear assertion comment that gives purpose of assertion, even if it's to specific at the moment. The main idea is to help in case of failure, and knowing that it fails because there is no chat window of thread is super helpful.

In the meantime of improving this test so that detection of opening of thread is not chat window-dependent, I prefer being super explicit when the assertion relies on that.